tests/functional/interfaces-with-arguments.ts

```diff
@@ -0,0 +1,406 @@
+      ) as IntrospectionInterfaceType).fields.find(
+        f => f.name === "implemetingObjectTypeFieldInlineArguments",
+      ) as IntrospectionField;
+
+      expect(sampleField.args).toBeDefined();
+      expect(sampleField.args.length).toEqual(2);
+      expect(
+        sampleField.args.every(arg => ["intValue1", "intValue2"].includes(arg.name)),
+      ).toBeTruthy();
+    });
+
+    it("should have proper arguments for the object field with an arguments type", async () => {
+      const sampleField = (schemaIntrospection.types.find(
+        type => type.name === "SampleImplementingObject1",
+      ) as IntrospectionInterfaceType).fields.find(
+        f => f.name === "implemetingObjectTypeFieldArgumentsType",
+      ) as IntrospectionField;
+
+      expect(sampleField.args).toBeDefined();
+      expect(sampleField.args.length).toEqual(2);
+      expect(
+        sampleField.args.every(arg => ["intValue1", "intValue2"].includes(arg.name)),
+      ).toBeTruthy();
+      expect(sampleField.args.find(a => a.name === "intValue1")!.defaultValue).toEqual("50");
+    });
+
+    it("should have proper arguments for the object field-resolver", async () => {
+      const sampleField = (schemaIntrospection.types.find(
+        type => type.name === "SampleImplementingObject1",
+      ) as IntrospectionInterfaceType).fields.find(
+        f => f.name === "sampleFieldResolver",
+      ) as IntrospectionField;
+
+      expect(sampleField.args).toBeDefined();
+      expect(sampleField.args.length).toEqual(2);
+      expect(
+        sampleField.args.every(arg => ["intValue1", "intValue2"].includes(arg.name)),
+      ).toBeTruthy();
+      expect(sampleField.args.find(a => a.name === "intValue1")!.defaultValue).toEqual("50");
+    });
+  });
+
+  describe("Functional", () => {
+    const resolvedValueString = (intValue1?: number, intValue2?: number) =>
+      `${intValue1}-${intValue2}`;
+    let schema: GraphQLSchema;
+
+    beforeAll(async () => {
+      getMetadataStorage().clear();
+
+      @ArgsType()
+      class SampleArgs1 {
+        @Field(type => Int, { nullable: true, defaultValue: 50 })
+        intValue1?: number;
+        @Field(type => Int, { nullable: true })
+        intValue2?: number;
+      }
+
+      @InterfaceType()
+      abstract class SampleInterface1 {
+        @Field(type => ID)
+        id: string;
+        @Field({ nullable: true })
+        interfaceStringField1?: string;
+
+        @Field()
+        interfaceFieldInlineArguments(
+          @Arg("intValue1") intValue1: number,
+          @Arg("intValue2") intValue2: number,
+        ): string {
+          throw new Error("Method not implemented.");
+        }
+        @Field()
+        interfaceFieldArgumentsType(@Args() args: SampleArgs1): string {
+          throw new Error("Method not implemented.");
+        }
+      }
+
+      @ObjectType({ implements: SampleInterface1 })
+      class SampleImplementingObject1 implements SampleInterface1 {
```

Comment:
The `ObjectType extends InterfaceType` is a nice pattern to reduce the boilerplate of redeclaring fields, we should support that too as it's handy with sharing implementation of interface field resolver. So please test that behavior in both schema and functional way and also check overriding the inline field resolvers methods from base interface.